app-backend/db/src/main/scala/TaskDao.scala

```diff
@@ -294,4 +294,111 @@ object TaskDao extends Dao[Task] {
   def deleteLayerTasks(projectId: UUID, layerId: UUID): ConnectionIO[Int] = {
     (fr"DELETE FROM " ++ this.tableF ++ fr"WHERE project_id = ${projectId} and project_layer_id = ${layerId}").update.run
   }
+
+  def getTaskActionTimeF(projectId: UUID,
+                         layerId: UUID,
+                         fromStatus: TaskStatus,
+                         toStatus: TaskStatus,
+                         params: UserTaskActivityParameters): Fragment = {
+    val selectTaskActionF =
+      fr"SELECT task_actions.user_id, tasks.id as task_id, MAX(task_actions.timestamp) AS timestamp FROM"
+    val joinTaskActionF =
+      fr"tasks LEFT JOIN task_actions ON task_actions.task_id = tasks.id"
+    val whereClause = Fragments
+      .whereAndOpt(
+        Some(fr"project_id = uuid($projectId)"),
+        Some(fr"project_layer_id = uuid($layerId)"),
+        Some(fr"task_actions.FROM_status = $fromStatus"),
```

Comment:
should be lower case - postgres casts everything to lower case unless it's in double quotes.